--- linux-4.16.2/kernel/time/tick-sched.c~	2018-04-12 10:30:01.000000000 +0000
+++ linux-4.16.2/kernel/time/tick-sched.c	2018-04-17 16:29:30.184622065 +0000
@@ -743,8 +743,8 @@
 
 #ifdef CONFIG_NO_HZ_FULL
 	/* Limit the tick delta to the maximum scheduler deferment */
-	if (!ts->inidle)
-		delta = min(delta, scheduler_tick_max_deferment());
+//	if (!ts->inidle)
+//		delta = min(delta, scheduler_tick_max_deferment());
 #endif
 
 	/* Calculate the next expiry time */
