From 5a2cb154e11490e5ba05975c7a537d0bafb12bba Mon Sep 17 00:00:00 2001
From: Arjan van de Ven <arjan@linux.intel.com>
Date: Sun, 22 Jan 2017 18:51:13 +0000
Subject: [PATCH 122/126] tweak perfbias

---
 arch/x86/kernel/cpu/intel.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/x86/kernel/cpu/intel.c b/arch/x86/kernel/cpu/intel.c
index b720dacac051..c01bb1632cbf 100644
--- a/arch/x86/kernel/cpu/intel.c
+++ b/arch/x86/kernel/cpu/intel.c
@@ -467,12 +467,12 @@ static void init_intel_energy_perf(struct cpuinfo_x86 *c)
 		return;
 
 	rdmsrl(MSR_IA32_ENERGY_PERF_BIAS, epb);
-	if ((epb & 0xF) != ENERGY_PERF_BIAS_PERFORMANCE)
+	if ((epb & 0xF) >= ENERGY_PERF_BIAS_NORMAL)
 		return;
 
-	pr_warn_once("ENERGY_PERF_BIAS: Set to 'normal', was 'performance'\n");
+	pr_warn_once("ENERGY_PERF_BIAS: Set to 'performance', was 'normal'\n");
 	pr_warn_once("ENERGY_PERF_BIAS: View and update with x86_energy_perf_policy(8)\n");
-	epb = (epb & ~0xF) | ENERGY_PERF_BIAS_NORMAL;
+	epb = (epb & ~0xF) | ENERGY_PERF_BIAS_PERFORMANCE;
 	wrmsrl(MSR_IA32_ENERGY_PERF_BIAS, epb);
 }
 
-- 
2.15.0

