From df19d4564a6425b44d26c477e474881b1b573c46 Mon Sep 17 00:00:00 2001
From: Arjan van de Ven <arjan@linux.intel.com>
Date: Sat, 29 Apr 2017 22:24:34 +0000
Subject: [PATCH 124/126] kernel: time: reduce ntp wakeups

---
 kernel/time/ntp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/time/ntp.c b/kernel/time/ntp.c
index 99e03bec68e4..5183f012565c 100644
--- a/kernel/time/ntp.c
+++ b/kernel/time/ntp.c
@@ -559,7 +559,7 @@ static void sync_cmos_clock(struct work_struct *work)
 	if (!fail || fail == -ENODEV)
 		next.tv_sec = 659;
 	else
-		next.tv_sec = 0;
+		next.tv_sec = 10;
 
 	if (next.tv_nsec >= NSEC_PER_SEC) {
 		next.tv_sec++;
-- 
2.15.0

