diff -purN linux-4.11.6.org/include/linux/sched.h linux-4.11.6/include/linux/sched.h
--- linux-4.11.6.org/include/linux/sched.h	2017-06-17 04:47:27.000000000 +0000
+++ linux-4.11.6/include/linux/sched.h	2017-06-20 20:02:36.943110101 +0000
@@ -703,6 +703,7 @@ struct task_struct {
 	struct task_cputime		cputime_expires;
 	struct list_head		cpu_timers[3];
 #endif
+	int fsync_count;
 
 	/* Process credentials: */
 
diff -purN linux-4.11.6.org/kernel/sched/debug.c linux-4.11.6/kernel/sched/debug.c
--- linux-4.11.6.org/kernel/sched/debug.c	2017-06-17 04:47:27.000000000 +0000
+++ linux-4.11.6/kernel/sched/debug.c	2017-06-20 20:16:29.280717474 +0000
@@ -884,6 +884,7 @@ void proc_sched_show_task(struct task_st
 	PN(se.exec_start);
 	PN(se.vruntime);
 	PN(se.sum_exec_runtime);
+	P(fsync_count);
 
 	nr_switches = p->nvcsw + p->nivcsw;
 
--- linux-4.11/block/blk-core.c~	2017-05-01 02:47:48.000000000 +0000
+++ linux-4.11/block/blk-core.c	2017-06-21 03:30:33.754638378 +0000
@@ -2087,6 +2087,9 @@
 			count_vm_events(PGPGIN, count);
 		}
 
+		if (bio->bi_opf & REQ_PREFLUSH)
+			current->fsync_count++;
+
 		if (unlikely(block_dump)) {
 			char b[BDEVNAME_SIZE];
 			printk(KERN_DEBUG "%s(%d): %s block %Lu on %s (%u sectors)\n",
