From 92fba5bb6c935ce18405b4581aa001bf592fd6dc Mon Sep 17 00:00:00 2001
From: Miguel Bernal Marin <miguel.bernal.marin@linux.intel.com>
Date: Thu, 5 May 2016 22:34:55 -0500
Subject: [PATCH 1002/1002] dpdk: integrate Kconfig and Makefiles

Signed-off-by: Miguel Bernal Marin <miguel.bernal.marin@linux.intel.com>
---
 drivers/net/Kconfig  | 2 ++
 drivers/net/Makefile | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/drivers/net/Kconfig b/drivers/net/Kconfig
index 0c5415b..33f9bbb 100644
--- a/drivers/net/Kconfig
+++ b/drivers/net/Kconfig
@@ -449,4 +449,6 @@ config FUJITSU_ES
 
 source "drivers/net/hyperv/Kconfig"
 
+source "drivers/net/dpdk/Kconfig"
+
 endif # NETDEVICES
diff --git a/drivers/net/Makefile b/drivers/net/Makefile
index 7336cbd..1369c08 100644
--- a/drivers/net/Makefile
+++ b/drivers/net/Makefile
@@ -72,3 +72,5 @@ obj-$(CONFIG_HYPERV_NET) += hyperv/
 obj-$(CONFIG_NTB_NETDEV) += ntb_netdev.o
 
 obj-$(CONFIG_FUJITSU_ES) += fjes/
+
+obj-$(CONFIG_DPDK) += dpdk/
-- 
2.9.1

