From 25fe4baa7d8f0ba97e932b08307fa95ede635026 Mon Sep 17 00:00:00 2001
From: Miguel Bernal Marin <miguel.bernal.marin@linux.intel.com>
Date: Thu, 5 May 2016 22:34:55 -0500
Subject: [PATCH 2002/2002] dpdk: integrate Kconfig and Makefiles

Signed-off-by: Miguel Bernal Marin <miguel.bernal.marin@linux.intel.com>
---
 drivers/net/Kconfig  | 2 ++
 drivers/net/Makefile | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/drivers/net/Kconfig b/drivers/net/Kconfig
index befd67d..880f134 100644
--- a/drivers/net/Kconfig
+++ b/drivers/net/Kconfig
@@ -432,4 +432,6 @@ config FUJITSU_ES
 
 source "drivers/net/hyperv/Kconfig"
 
+source "drivers/net/dpdk/Kconfig"
+
 endif # NETDEVICES
diff --git a/drivers/net/Makefile b/drivers/net/Makefile
index 1aa7cb8..887a743 100644
--- a/drivers/net/Makefile
+++ b/drivers/net/Makefile
@@ -71,3 +71,5 @@ obj-$(CONFIG_HYPERV_NET) += hyperv/
 obj-$(CONFIG_NTB_NETDEV) += ntb_netdev.o
 
 obj-$(CONFIG_FUJITSU_ES) += fjes/
+
+obj-$(CONFIG_DPDK) += dpdk/
-- 
2.8.2

