From d7eb7ac9e515c25d4198bd89f36ae76075fd6058 Mon Sep 17 00:00:00 2001
From: Miguel Bernal Marin <miguel.bernal.marin@linux.intel.com>
Date: Fri, 25 Sep 2015 19:27:52 -0500
Subject: [PATCH 5002/5002] dpdk: Integrate Kconfig and Makefiles

Signed-off-by: Miguel Bernal Marin <miguel.bernal.marin@linux.intel.com>
---
 drivers/net/Kconfig  | 2 ++
 drivers/net/Makefile | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/drivers/net/Kconfig b/drivers/net/Kconfig
index c18f9e6..490fe01 100644
--- a/drivers/net/Kconfig
+++ b/drivers/net/Kconfig
@@ -409,4 +409,6 @@ config VMXNET3
 
 source "drivers/net/hyperv/Kconfig"
 
+source "drivers/net/dpdk/Kconfig"
+
 endif # NETDEVICES
diff --git a/drivers/net/Makefile b/drivers/net/Makefile
index c12cb22..6c274e9 100644
--- a/drivers/net/Makefile
+++ b/drivers/net/Makefile
@@ -67,3 +67,5 @@ obj-$(CONFIG_USB_NET_DRIVERS) += usb/
 
 obj-$(CONFIG_HYPERV_NET) += hyperv/
 obj-$(CONFIG_NTB_NETDEV) += ntb_netdev.o
+
+obj-$(CONFIG_DPDK) += dpdk/
-- 
2.4.3

