--- linux-4.5/drivers/input/serio/i8042.c~	2016-03-14 04:28:54.000000000 +0000
+++ linux-4.5/drivers/input/serio/i8042.c	2016-04-18 19:01:44.626983477 +0000
@@ -25,6 +25,7 @@
 #include <linux/i8042.h>
 #include <linux/slab.h>
 #include <linux/suspend.h>
+#include <linux/async.h>
 
 #include <asm/io.h>
 
@@ -1593,6 +1594,44 @@
 	return err;
 }
 
+static int i8042_init_async(void *data, async_cookie_t cookie)
+{
+	struct platform_device *pdev;
+	int err;
+
+	dbg_init();
+
+	err = i8042_platform_init();
+	if (err)
+		return err;
+
+	err = i8042_controller_check();
+	if (err)
+		goto err_platform_exit;
+
+	pdev = platform_create_bundle(&i8042_driver, i8042_probe, NULL, 0, NULL, 0);
+	if (IS_ERR(pdev)) {
+		err = PTR_ERR(pdev);
+		goto err_platform_exit;
+	}
+
+	bus_register_notifier(&serio_bus, &i8042_kbd_bind_notifier_block);
+	panic_blink = i8042_panic_blink;
+
+	return 0;
+
+ err_platform_exit:
+	i8042_platform_exit();
+	return err;
+}
+
+static __init int i8042_init(void)
+{
+	async_schedule(i8042_init_async, NULL);
+	return 0;
+}
+
+
 static void __exit i8042_exit(void)
 {
 	platform_device_unregister(i8042_platform_device);
