--- linux-5.3.7/arch/x86/Makefile~	2019-10-17 20:47:33.000000000 +0000
+++ linux-5.3.7/arch/x86/Makefile	2019-11-06 03:54:54.550351296 +0000
@@ -208,6 +208,9 @@
 ifdef CONFIG_X86_64
 KBUILD_LDFLAGS += $(call ld-option, -z max-page-size=0x200000)
 endif
+ifdef CONFIG_X86_64
+KBUILD_LDFLAGS += $(call ld-option, -mbranches-within-no-boundaries)
+endif
 
 # Workaround for a gcc prelease that unfortunately was shipped in a suse release
 KBUILD_CFLAGS += -Wno-sign-compare
--- linux-5.3.7/arch/x86/Makefile~	2019-11-06 03:54:54.000000000 +0000
+++ linux-5.3.7/arch/x86/Makefile	2019-11-06 04:00:55.395346272 +0000
@@ -58,7 +58,7 @@
 #
 #    https://gcc.gnu.org/bugzilla/show_bug.cgi?id=53383
 #
-KBUILD_CFLAGS += -mno-sse -mno-mmx -mno-sse2 -mno-3dnow
+KBUILD_CFLAGS += -mno-sse -mno-mmx -mno-sse2 -mno-3dnow -Wa,-mbranches-within-no-boundaries 
 KBUILD_CFLAGS += $(call cc-option,-mno-avx,)
 
 ifeq ($(CONFIG_X86_32),y)
--- linux-5.3.7/arch/x86/Makefile~	2019-11-06 04:00:55.000000000 +0000
+++ linux-5.3.7/arch/x86/Makefile	2019-11-06 04:16:17.063333440 +0000
@@ -195,8 +195,8 @@
 sha1_ni_instr :=$(call as-instr,sha1msg1 %xmm0$(comma)%xmm1,-DCONFIG_AS_SHA1_NI=1)
 sha256_ni_instr :=$(call as-instr,sha256msg1 %xmm0$(comma)%xmm1,-DCONFIG_AS_SHA256_NI=1)
 
-KBUILD_AFLAGS += $(cfi) $(cfi-sigframe) $(cfi-sections) $(asinstr) $(avx_instr) $(avx2_instr) $(avx512_instr) $(sha1_ni_instr) $(sha256_ni_instr)
-KBUILD_CFLAGS += $(cfi) $(cfi-sigframe) $(cfi-sections) $(asinstr) $(avx_instr) $(avx2_instr) $(avx512_instr) $(sha1_ni_instr) $(sha256_ni_instr)
+KBUILD_AFLAGS += $(cfi) $(cfi-sigframe) $(cfi-sections) $(asinstr) $(avx_instr) $(avx2_instr) $(avx512_instr) $(sha1_ni_instr) $(sha256_ni_instr) -Wa,-mbranches-within-no-boundaries
+KBUILD_CFLAGS += $(cfi) $(cfi-sigframe) $(cfi-sections) $(asinstr) $(avx_instr) $(avx2_instr) $(avx512_instr) $(sha1_ni_instr) $(sha256_ni_instr) -Wa,-mbranches-within-no-boundaries
 
 KBUILD_LDFLAGS := -m elf_$(UTS_MACHINE)
 
--- linux-5.3.7/arch/x86/boot/compressed/Makefile~	2019-10-17 20:47:33.000000000 +0000
+++ linux-5.3.7/arch/x86/boot/compressed/Makefile	2019-11-06 04:23:47.533327168 +0000
@@ -39,7 +39,7 @@
 KBUILD_CFLAGS += $(call cc-disable-warning, gnu)
 KBUILD_CFLAGS += -Wno-pointer-sign
 
-KBUILD_AFLAGS  := $(KBUILD_CFLAGS) -D__ASSEMBLY__
+KBUILD_AFLAGS  := $(KBUILD_CFLAGS) -D__ASSEMBLY__  -Wa,-mbranches-within-no-boundaries
 GCOV_PROFILE := n
 UBSAN_SANITIZE :=n
 
