these error messages are causing support isseus

--- linux-6.1/arch/x86/kvm/x86.c~	2022-12-11 22:15:18.000000000 +0000
+++ linux-6.1/arch/x86/kvm/x86.c	2022-12-15 22:15:07.085648692 +0000
@@ -9340,13 +9340,9 @@
 	}
 
 	if (!ops->cpu_has_kvm_support()) {
-		pr_err_ratelimited("kvm: no hardware support for '%s'\n",
-				   ops->runtime_ops->name);
 		return -EOPNOTSUPP;
 	}
 	if (ops->disabled_by_bios()) {
-		pr_err_ratelimited("kvm: support for '%s' disabled by bios\n",
-				   ops->runtime_ops->name);
 		return -EOPNOTSUPP;
 	}
 
