--- linux-5.3.7/drivers/nvme/host/core.c~	2019-10-17 20:47:33.000000000 +0000
+++ linux-5.3.7/drivers/nvme/host/core.c	2019-11-11 23:11:06.202370793 +0000
@@ -48,7 +48,7 @@
 module_param_named(max_retries, nvme_max_retries, byte, 0644);
 MODULE_PARM_DESC(max_retries, "max number of retries a command may have");
 
-static unsigned long default_ps_max_latency_us = 100000;
+static unsigned long default_ps_max_latency_us = 200;
 module_param(default_ps_max_latency_us, ulong, 0644);
 MODULE_PARM_DESC(default_ps_max_latency_us,
 		 "max power saving latency for new devices; use PM QOS to change per device");
