from https://github.com/jplozi/wastedcores
diff -uprNw -X linux-4.1.vanilla/Documentation/dontdiff linux-4.1.vanilla/kernel/sched/fair.c linux-4.1.overload-on-wakeup/kernel/sched/fair.c
--- linux-4.1.vanilla/kernel/sched/fair.c	2015-06-21 22:05:43.000000000 -0700
+++ linux-4.1.overload-on-wakeup/kernel/sched/fair.c	2015-11-05 01:30:19.693493606 -0800
@@ -4834,6 +4834,20 @@ select_task_rq_fair(struct task_struct *
 		want_affine = !wake_wide(p) && cpumask_test_cpu(cpu, tsk_cpus_allowed(p));
 
 	rcu_read_lock();
+
+	if (cpu_rq(prev_cpu)->nr_running)
+	{
+		int _cpu;
+		for_each_online_cpu(_cpu)
+		{
+			if (!cpumask_test_cpu(_cpu, tsk_cpus_allowed(p)) ||
+			            cpu_rq(_cpu)->nr_running) continue;
+
+			rcu_read_unlock();
+			return _cpu; 
+		}
+	}
+
 	for_each_domain(cpu, tmp) {
 		if (!(tmp->flags & SD_LOAD_BALANCE))
 			break;
