--- linux-6.10.11/mm/mmap.c~	2024-09-18 17:25:18.000000000 +0000
+++ linux-6.10.11/mm/mmap.c	2024-09-26 23:45:20.399101364 +0000
@@ -1881,10 +1881,11 @@
 
 	if (get_area) {
 		addr = get_area(file, addr, len, pgoff, flags);
-	} else if (IS_ENABLED(CONFIG_TRANSPARENT_HUGEPAGE)) {
+//	} 
+//else if (IS_ENABLED(CONFIG_TRANSPARENT_HUGEPAGE)) {
 		/* Ensures that larger anonymous mappings are THP aligned. */
-		addr = thp_get_unmapped_area_vmflags(file, addr, len,
-						     pgoff, flags, vm_flags);
+		//addr = thp_get_unmapped_area_vmflags(file, addr, len,
+//						     pgoff, flags, vm_flags);
 	} else {
 		addr = mm_get_unmapped_area_vmflags(current->mm, file, addr, len,
 						    pgoff, flags, vm_flags);
