--- linux-4.15.2/kernel/locking/rwsem-xadd.c~	2018-02-07 19:14:15.000000000 +0000
+++ linux-4.15.2/kernel/locking/rwsem-xadd.c	2018-02-18 23:34:48.457758693 +0000
@@ -381,6 +381,7 @@
 static noinline bool rwsem_spin_on_owner(struct rw_semaphore *sem)
 {
 	struct task_struct *owner = READ_ONCE(sem->owner);
+	int i = 0;
 
 	if (!rwsem_owner_is_writer(owner))
 		goto out;
@@ -404,8 +405,8 @@
 			rcu_read_unlock();
 			return false;
 		}
-
-		cpu_relax();
+		if (i++ > 1000)
+			cpu_relax();
 	}
 	rcu_read_unlock();
 out:
