--- linux-4.15.13/arch/x86/Makefile~	2018-03-24 10:02:53.000000000 +0000
+++ linux-4.15.13/arch/x86/Makefile	2018-03-28 14:58:17.263227363 +0000
@@ -233,7 +233,7 @@
 # Avoid indirect branches in kernel to deal with Spectre
 ifdef CONFIG_RETPOLINE
 ifneq ($(RETPOLINE_CFLAGS),)
-  KBUILD_CFLAGS += $(RETPOLINE_CFLAGS) -DRETPOLINE
+  KBUILD_CFLAGS += $(RETPOLINE_CFLAGS) -DRETPOLINE -mzero-caller-saved-regs
 endif
 endif
 
